pyuvdata/uvcal/uvcal.py

```diff
@@ -3351,6 +3355,446 @@ def _convert_to_filetype(self, filetype):
+            if uvdata_int_times.size == 1:
+                uvdata_int_times = uvdata_int_times[0]
+                if uvc.future_array_shapes:
+                    uvc.integration_time = np.full(
+                        uvc.time_array.size, uvdata_int_times, dtype=np.float64
+                    )
+                else:
+                    uvc.integration_time = uvdata_int_times
+            else:
+                raise ValueError(
+                    "uvdata integration times vary. Please specify times and "
+                    "integration_time"
+                )
+        else:
+            uvc.time_array = times
+            if integration_time is None:
+                raise ValueError(
+                    "integation_time must be provided if times is provided"
+                )
+            if future_array_shapes:
+                if isinstance(integration_time, (np.ndarray, list)):
+                    uvc.integration_time = np.asarray(integration_time)
+                else:
+                    uvc.integration_time = np.full(
+                        uvc.time_array.size, integration_time, dtype=np.float64
+                    )
+            else:
+                if isinstance(integration_time, (np.ndarray, list)):
+                    raise ValueError(
+                        "integration_time must be scalar if future_array_shapes is "
+                        "False."
+                    )
+                uvc.integration_time = integration_time
+        uvc.Ntimes = uvc.time_array.size
+        uvc.set_lsts_from_time_array()
+
+        if time_range is not None:
+            uvc.time_range = time_range
+
+        if jones is None:
+            if np.all(uvdata.polarization_array < -4):
+                if uvdata.Npols == 1 and uvdata.polarization_array[0] > -7:
+                    # single pol data, make a single pol cal object
+                    uvc.jones_array = uvdata.polarization_array
+                else:
+                    uvc.jones_array = np.array([-5, -6])
+            elif np.all(uvdata.polarization_array < 0):
+                if uvdata.Npols == 1 and uvdata.polarization_array[0] > -3:
+                    # single pol data, make a single pol cal object
+                    uvc.jones_array = uvdata.polarization_array
+                else:
+                    uvc.jones_array = np.array([-1, -2])
+            else:
+                raise ValueError(
+                    "jones parameter is None and uvdata object is in "
+                    "psuedo-stokes polarization. Please set jones."
+                )
+        else:
+            uvc.jones_array = np.asarray(jones)
+        uvc.Njones = uvc.jones_array.size
+
+        uvc.ant_array = np.union1d(uvdata.ant_1_array, uvdata.ant_2_array)
+
+        uvc.history = "Initialized from a UVData object with pyuvdata."
+        if include_uvdata_history:
+            uvc.history += " UVData history is: " + uvdata.history
+
+        if not metadata_only:
```

Comment:
I think "initialize with zeros" is the correct default. But maybe we want to make this explicit to the user in the docstring? (I could imagine the user potentially thinking the flag array might be transferred, for instance...)

Reply:
I have this in the docstring under the `metadata_only` parameter:
```
        metadata_only : bool
            Option to only initialize the metadata. If False, this method also
            initializes the data-like arrays to zeros (or False for the flag_array)
            with the appropriate sizes.
```
Should I add something at the top level? The default is `metadata_only=True`.

Reply:
Ah thanks, didn't realize it was in the parameter docstring. Because the default is `metadata_only=True`, then I don't think we need extra documentation beyond this.